rpc/rpc-transport/socket/src/socket.c

```diff
@@ -3217,12 +3217,21 @@ socket_connect(rpc_transport_t *this, int port)
     pthread_mutex_lock(&priv->out_lock);
     {
         if (priv->sock >= 0) {
-            gf_log_callingfn(this->name, GF_LOG_TRACE,
-                             "connect () called on transport "
-                             "already connected");
-            errno = EINPROGRESS;
-            ret = -1;
-            goto unlock;
+            if (priv->connected == 0) {
+                /* shutdown(2) will result in EPOLLERR, so cleanup is done
+                 * in socket_event_handler
+                 */
+                shutdown(priv->sock, SHUT_RDWR);
```

Comment:
I think in case of calling shutdown either we should return ret with 0 or return shutdown return code to the upper layer.

Reply:
thank you